community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1792,7 +1811,49 @@ def nodeset_reservation(self, nodeset: object) -> Optional[ReservationDetails]:
             zone=zone,
             name=name,
             policies=policies,
-            bulk_insert_name=nodeset.reservation_name)
+            bulk_insert_name=reservation_name)
+    
+    def future_reservation(self, futureReservation:str, zone_policy_allow: Optional[list]) -> Optional[FutureReservation]:
+        zones = list(zone_policy_allow or [])
+        assert len(zones) == 1, "Only single zone is supported if using a reservation"
+        zone = zones[0]
+        
+        project = self.project
+        name = futureReservation
+        nodesAreActive = False
+        specificReservationRequired = None
+        activeReservation = None
+        endTime = None
+        startTime = None
+        
+        regex = re.compile(r'/futureReservations/(?P<name>[^/]+)(/.*)?$')
+
+        if match := regex.search(futureReservation):
```

Comment:
`regex` used once, compiling it doesn't help to performance, you can save some lines:
```py
if match := re.search(r'/futureReservations/(?P<name>[^/]+)(/.*)?$', futureReservation)
```